app/views/reports/regions/_child_data_tables_with_exclusions.html.erb

```diff
@@ -57,64 +57,61 @@
           <td class="ta-left">
-            <%= number_with_delimiter(@data.last_value(:cumulative_registrations)) %>
+            <%= number_with_delimiter(@data.dig(:cumulative_registrations, @period)) %>
           </td>
           <td class="ta-left">
-            <%= number_with_delimiter(@data.last_value(:registrations)) %>
+            <%= number_with_delimiter(@data.dig(:registrations, @period)) %>
           </td>
         </tr>
         <% data.each do |result| %>
-          <% next if result.dig("controlled_patients_rate", @period).nil? %>
-          <% next if result.dig("uncontrolled_patients_rate", @period).nil? %>
-          <% next if result.dig("missed_visits_rate", @period).nil? %>
-          <% next if result.dig("registrations", @period).nil? %>
+          <% next if result.dig(:missed_visits_rate, @period).nil? %>
           <tr>
             <td class="ta-left">
               <%= link_to(reports_region_path(result.region, report_scope: region_type)) do %>
                 <%= result.region.name %>
               <% end %>
             </td>
             <td class="ta-left">
-              <%= number_to_percentage(result["controlled_patients_rate"].values.last || 0, precision: 0) %>
+              <%= number_to_percentage(result.dig(:controlled_patients_rate, @period), precision: 0) %>
             </td>
             <td class="ta-left">
-              <%= number_with_delimiter(result["controlled_patients"].values.last) %>
+              <%= number_with_delimiter(result.dig(:controlled_patients, @period)) %>
             </td>
             <td class="ta-left">
-              <%= number_to_percentage(result["uncontrolled_patients_rate"].values.last || 0, precision: 0) %>
+              <%= number_to_percentage(result.dig(:uncontrolled_patients_rate, @period), precision: 0) %>
             </td>
             <td class="ta-left">
-              <%= number_with_delimiter(result["uncontrolled_patients"].values.last) %>
+              <%= number_with_delimiter(result.dig(:uncontrolled_patients, @period)) %>
             </td>
             <td class="ta-left">
-              <%= number_to_percentage(result["missed_visits_rate"].values.last || 0, precision: 0) %>
+              <%= number_to_percentage(result.dig(:missed_visits_rate, @period), precision: 0) %>
             </td>
             <td class="ta-left">
-              <%= number_with_delimiter(result["missed_visits"].values.last) %>
+              <%= number_with_delimiter(result.dig(:missed_visits, @period)) %>
             </td>
             <td class="ta-left">
-              <%= number_with_delimiter(result["cumulative_registrations"].values.last) %>
+              <%= number_with_delimiter(result.dig(:cumulative_registrations, @period)) %>
             </td>
             <td class="ta-left">
-              <%= number_with_delimiter(result["registrations"].values.last || 0) %>
+              <%= number_with_delimiter(result.dig(:registrations, @period)) %>
```

Comment:
This line is the actual bug, but the problem fundamentally is that we are relying on positional data when the positions aren't actually pinned to the current month, which is why I've changed all of the positional data fetching in this file. I think a better fix would be to pin the queries to the requested period more firmly because this bug is likely to pop up elsewhere otherwise. But that is outside the scope of the current bug fix and would possibly cause bugs elsewhere if done without sufficient care.